src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/MatchXpathCheckTest.java

```diff
@@ -91,6 +91,31 @@ public void testCheckWithNoMatchingMethodName()
         verify(checkConfig, getPath("InputMatchXpath.java"), expected);
     }
 
+    @Test
+    public void testCheckWithSingleLineComments() throws Exception {
+        final DefaultConfiguration checkConfig =
+                createModuleConfig(MatchXpathCheck.class);
+        checkConfig.addAttribute("query", "//SINGLE_LINE_COMMENT");
+        final String[] expected = {
+            "11:9: " + getCheckMessage(MatchXpathCheck.MSG_KEY),
+            "12:18: " + getCheckMessage(MatchXpathCheck.MSG_KEY),
+        };
+        verify(checkConfig, getPath("InputMatchXpathSingleLineComments.java"), expected);
+    }
+
+    @Test
+    public void testCheckWithBlockComments() throws Exception {
+        final DefaultConfiguration checkConfig =
+                createModuleConfig(MatchXpathCheck.class);
+        checkConfig.addAttribute("query", "//BLOCK_COMMENT_BEGIN");
```

Comment:
ability to violate a token do not have real life value, please update query to use text of comment.
Attention: please show how it can work on matching multiline comments (where first part bad pattern on one line and second part on other line).

Reply:
> can we detect that multiline comment present inside methods (good practice is to use only single line comments in methods)

Sure we can, the XPath will be similar to `//METHOD_DEF[//*]/SLIST//*/BLOCK_COMMENT_BEGIN`, but I believe it is not required to show as it each example requires a different file, and this is not a complex example.

The other two examples have been added as test cases.